src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -227,9 +233,25 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
         }
 
         try {
-            return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
+            return [$this->docBlockFactory->create($reflectionMethod, $this->createFromReflector($reflectionMethod->getDeclaringClass())), $prefix];
         } catch (\InvalidArgumentException $e) {
             return null;
         }
     }
+
+    /**
+     * Prevents a lot of redundant calls to ContextFactory::createForNamespace().
+     */
+    private function createFromReflector(\ReflectionClass $reflector): Context
+    {
+        $cacheKey = $reflector->getNamespaceName().':'.$reflector->getFileName();
```

Comment:
Couldn't this still create conflicting namespace keys if two classes are defined in the same file?

Reply:
not really, see the implementation here: https://github.com/phpDocumentor/TypeResolver/blob/master/src/Types/ContextFactory.php#L87